api/v1/controllers/TeamController.js

```diff
@@ -0,0 +1,115 @@
+var errors = require('../errors');
+var services = require('../services');
+var roles = require('../utils/roles');
+
+var bodyParser = require('body-parser');
+var middleware = require('../middleware');
+var router = require('express').Router();
+
+function joinTeam (req, res, next) {
+  var email = req.auth.email;
+  var teamName = req.body.teamName;
+
+  return services.UserService.findUserByEmail(email)
+  .then(function(user){
+	return services.TeamService
+	  .addUserToTeam(user, teamName);
+  })
+  .then(function(team){
```

Comment:
Well my note disappeared, so in case:
Any reasons as to why you are using this instead of
```javascript
then(function(team){
    res.body = team.toJSON();
    next();
    return null;
})